docs/02-guides/maintenance-mode/automatic-detection-with-service-health-checks.mdx

```diff
@@ -0,0 +1,162 @@
+---
+title: Automatic detection with service Health Checks
+description:
+  Learn how to implement health checks to detect down services and automatically
+  place your site in maintenance mode.
+sidebar_position: 2
+---
+
+<SinceVersion tag="3.6" />
+
+# Health Checks
+
+Health checks allow you to monitor the availability of your services and
+automatically put your store in maintenance mode if a service is down. This
+ensures that your users are not affected by a service outage.
+
+With health checks, you can define a set of services that you want to monitor
+and provide a function to check the health of each service. The maintenance mode
+API will periodically run these health check functions and if a service is down,
+it will automatically put the store in maintenance mode. Once the service is
+back up, the maintenance mode API will automatically remove the store from
+maintenance mode.
```

Comment:
**nitpick:** at this level, I don't think mentioning the "Maintenance mode API" is relevant.

**suggestion:** use "Front-Commerce", which is more generic and (thus) futureproof

```suggestion
and provide a function to check the health of each service. Front-Commerce
will periodically run these health check functions and if a service is down,
it will automatically put the store in maintenance mode. Once the service is
back up, the Front-Commerce will automatically remove the store from
maintenance mode.
```